src/components/UploadMemeComponent.js

```diff
@@ -1,7 +1,10 @@
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form'
 import { useState } from 'react';
 import { useDispatch, useSelector } from "react-redux";
 
-import ShowSuccessMessage from "./show-success-component"
+import ShowSuccessMessage from "./ShowSuccessComponent"
+
 
 export default function UploadMeme() {
     const dispatch = useDispatch();
```

Comment:
`const currentHigherId = useSelector(state => Object.keys(state).reduce((a, b) => state[a] > state[b] ? a : b));`

true is that, you dont need `id` in your memes. you should have collection (Array of memes)
```
[
{ meme1 }
{ meme2 }
...
]
```
Array is sorting, and better to iterate. 
In your case `id` is unusable